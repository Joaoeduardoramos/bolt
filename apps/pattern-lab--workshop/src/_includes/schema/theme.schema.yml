"$schema": http://json-schema.org/draft-04/schema#
title: Theme
description: Core themes.
type: string
enum:
  - xlight
  - light
  - dark
  - xdark
  - medium
  - 'false'