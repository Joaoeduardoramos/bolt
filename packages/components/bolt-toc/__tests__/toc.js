import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
} from '../../../testing/testing-helpers';
import schema from '../toc.schema';
const { status, align } = schema.properties;
const timeout = 120000;

describe('<bolt-toc> Component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  // Basic Usage
  test('Basic usage', async () => {
    const results = await render('@bolt-components-toc/toc.twig', {
      items: [
        {
          content: 'Section One',
          url: '#section-one',
        },
        {
          content: 'Section Two',
          url: '#section-two',
        },
        {
          content: 'Section Three',
          url: '#section-three',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  test(`Table of content with a header`, async () => {
    const results = await render('@bolt-components-toc/toc.twig', {
      header: 'This is the header',
      items: [
        {
          content: 'Section One',
          url: '#section-one',
        },
        {
          content: 'Section Two',
          url: '#section-two',
        },
        {
          content: 'Section Three',
          url: '#section-three',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test(`Manuualy set the current item`, async () => {
    const results = await render('@bolt-components-toc/toc.twig', {
      items: [
        {
          content: 'Section One',
          url: '#section-one',
        },
        {
          content: 'Section Two',
          url: '#section-two',
          active: true,
        },
        {
          content: 'Section Three',
          url: '#section-three',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
