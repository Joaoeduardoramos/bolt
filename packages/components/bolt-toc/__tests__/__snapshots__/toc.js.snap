// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-toc> Component Basic usage 1`] = `
<bolt-toc>
  <nav class="c-bolt-toc"
       aria-labelledby="js-bolt-toc-12345"
  >
    <h2 class="c-bolt-toc__header c-bolt-toc__header--hidden"
        id="js-bolt-toc-12345"
    >
      Table of Content
    </h2>
    <ssr-keep for="bolt-toc"
              class="c-bolt-toc__list"
              role="list"
    >
      <bolt-toc-item url="#section-one"
                     role="presentation"
      >
        <replace-with-grandchildren role="listitem">
          <a class="c-bolt-toc-item "
             href="#section-one"
          >
            Section One
          </a>
        </replace-with-grandchildren>
      </bolt-toc-item>
      <bolt-toc-item url="#section-two"
                     role="presentation"
      >
        <replace-with-grandchildren role="listitem">
          <a class="c-bolt-toc-item "
             href="#section-two"
          >
            Section Two
          </a>
        </replace-with-grandchildren>
      </bolt-toc-item>
      <bolt-toc-item url="#section-three"
                     role="presentation"
      >
        <replace-with-grandchildren role="listitem">
          <a class="c-bolt-toc-item "
             href="#section-three"
          >
            Section Three
          </a>
        </replace-with-grandchildren>
      </bolt-toc-item>
    </ssr-keep>
  </nav>
</bolt-toc>
`;

exports[`<bolt-toc> Component Manuualy set the current item 1`] = `
<bolt-toc>
  <nav class="c-bolt-toc"
       aria-labelledby="js-bolt-toc-12345"
  >
    <h2 class="c-bolt-toc__header c-bolt-toc__header--hidden"
        id="js-bolt-toc-12345"
    >
      Table of Content
    </h2>
    <ssr-keep for="bolt-toc"
              class="c-bolt-toc__list"
              role="list"
    >
      <bolt-toc-item url="#section-one"
                     role="presentation"
      >
        <replace-with-grandchildren role="listitem">
          <a class="c-bolt-toc-item "
             href="#section-one"
          >
            Section One
          </a>
        </replace-with-grandchildren>
      </bolt-toc-item>
      <bolt-toc-item url="#section-two"
                     active
                     role="presentation"
      >
        <replace-with-grandchildren role="listitem">
          <a class="c-bolt-toc-item c-bolt-toc-item--current"
             href="#section-two"
          >
            Section Two
          </a>
        </replace-with-grandchildren>
      </bolt-toc-item>
      <bolt-toc-item url="#section-three"
                     role="presentation"
      >
        <replace-with-grandchildren role="listitem">
          <a class="c-bolt-toc-item "
             href="#section-three"
          >
            Section Three
          </a>
        </replace-with-grandchildren>
      </bolt-toc-item>
    </ssr-keep>
  </nav>
</bolt-toc>
`;

exports[`<bolt-toc> Component Table of content with a header 1`] = `
<bolt-toc header="This is the header">
  <nav class="c-bolt-toc"
       aria-labelledby="js-bolt-toc-12345"
  >
    <h2 class="c-bolt-toc__header"
        id="js-bolt-toc-12345"
    >
      This is the header
    </h2>
    <ssr-keep for="bolt-toc"
              class="c-bolt-toc__list"
              role="list"
    >
      <bolt-toc-item url="#section-one"
                     role="presentation"
      >
        <replace-with-grandchildren role="listitem">
          <a class="c-bolt-toc-item "
             href="#section-one"
          >
            Section One
          </a>
        </replace-with-grandchildren>
      </bolt-toc-item>
      <bolt-toc-item url="#section-two"
                     role="presentation"
      >
        <replace-with-grandchildren role="listitem">
          <a class="c-bolt-toc-item "
             href="#section-two"
          >
            Section Two
          </a>
        </replace-with-grandchildren>
      </bolt-toc-item>
      <bolt-toc-item url="#section-three"
                     role="presentation"
      >
        <replace-with-grandchildren role="listitem">
          <a class="c-bolt-toc-item "
             href="#section-three"
          >
            Section Three
          </a>
        </replace-with-grandchildren>
      </bolt-toc-item>
    </ssr-keep>
  </nav>
</bolt-toc>
`;
